.github/workflows/ci.yml

```diff
@@ -1,76 +1,116 @@
-name: nf-core CI
 # This workflow runs the pipeline with the minimal test dataset to check that it completes without any syntax errors
+name: nf-core CI
 on:
   push:
     branches:
-      - dev
+      - "dev"
   pull_request:
+    branches:
+      - "dev"
+      - "master"
   release:
-    types: [published]
+    types:
+      - "published"
 
 env:
   NXF_ANSI_LOG: false
+  NFTEST_VER: "0.7.3"
 
 concurrency:
-  group: "${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}"
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
+  define_nxf_versions:
+    name: Choose nextflow versions to test against depending on target branch
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.nxf_versions.outputs.matrix }}
+    steps:
+      - id: nxf_versions
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" && "${{ github.base_ref }}" == "dev" && "${{ matrix.NXF_VER }}" != "latest-everything" ]]; then
+            echo matrix='["latest-everything"]' | tee -a $GITHUB_OUTPUT
+          else
+            echo matrix='["latest-everything", "22.10.1"]' | tee -a $GITHUB_OUTPUT
+          fi
+
   test:
     name: Run pipeline with test data
-    # Only run on push if this is the nf-core dev branch (merged PRs)
-    if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/ampliseq') }}"
+    needs: define_nxf_versions
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        NXF_VER:
-          - "22.10.1"
-          - "latest-everything"
+        NXF_VER: ${{ fromJson(needs.define_nxf_versions.outputs.matrix) }}
+        component:
+          - "tests/pipeline/misc/custom_reftax.nf.test --profile test_reftaxcustom"
+          - "tests/pipeline/misc/doubleprimers.nf.test --profile test_doubleprimers"
+          - "tests/pipeline/misc/fasta_input.nf.test --profile test_fasta"
+          - "tests/pipeline/misc/multi_seq.nf.test --profile test_multi"
+          - "tests/pipeline/paired_end/paired_end.nf.test --profile test"
+          - "tests/pipeline/paired_end/pe_iontorrent.nf.test --profile test_iontorrent"
+          - "tests/pipeline/paired_end/pe_novaseq.nf.test --profile test_novaseq"
+          - "tests/pipeline/paired_end/pe_pacbio_its.nf.test --profile test_pacbio_its"
+          - "tests/pipeline/single_end/single_end.nf.test --profile test_single"
```

Comment:
Why did you choose 9 out of 11 tests (excluding test_full)?
Is there any reason for renaming them? (not that I care too much about the names...)
And categorizing into misc/pe/se? (not important to me, just to know)

Reply:
Yes! you are right here. 
> 
    *.nf.test.snap hold md5sums that are checked for via *.nf.test with snapshot(<file>).match(<key>)?
    in *.nf.test new File(<file>).exists() "only" checks for existence but not checksums

1. The decision to test dev PRs with `latest-everything` was made to free up the github runners and test master PR with `22.10.1` and `latest`
2. I didn't know if `test_full` is part of the CI for a PR ? was it previously?
3. agree on test naming...will update them along with changelog
4. the categorization in misc/pe/se was to parallelize tests but we have this now covered in ci `matrix.components` so will move them all to `tests/pipeline`

Reply:
Thanks!
1. fine
2. sorry, no, `test_full` is not and should not run on PR! But 2 other profiles are missing. I checked again, the profiles that you use are complete when looking at the master branch. But dev has two more: [test_pplace.config](https://github.com/nf-core/ampliseq/blob/dev/conf/test_pplace.config) & [test_sintax.config](https://github.com/nf-core/ampliseq/blob/dev/conf/test_sintax.config)
3. thanks
4. alright

Reply:
oh missed those....will add them in

Reply:
also initially any file that couldn't generate a consistent md5 was moved to exists checks. However we have a helper function to filter lines from a file that go into the snapshot so we can get consistent md5 and also not miss out on imp files integrity checks. if you notice any such files that should be in the snapshot, let me know.

Reply:
> I have added in those remaining two tests.

thanks!

> I have tried to snapshot or check for file consistency in all these test profiles but, it is possible I might have missed some that might be important.

Hm, so how are those `*.nf.test.snap` & `*.nf.test` produced? Isnt there some sort of script that goes over the results folder? If yes, how something "important" could be missed? Also asking because I need to maintain that tests somehow (new/changed/reduced files in results)...

Reply:
> Isnt there some sort of script that goes over the results folder?

No there isn't one yet so, I added all from results folder manually. 

> how something "important" could be missed?

this is in the context of a imp file that didn't make it into the snapshot due to differing md5. however as I said we now have functions to filter lines and then create a snapshot so it gets consistent md5

> nf-test test tests/pipeline/test.nf.test --update-snapshot should do the trick?

Yes, that's it!

Reply:
Thanks, that clarifies it!